### PR TITLE
Export Label and LabelDescriptor to Stackdriver

### PR DESCRIPTION
**Pull request by songy23**
Fix https://github.com/census-instrumentation/opencensus-go/issues/145

### CODE REVIEW

File: exporter/stats/stackdriver/stackdriver.go [hunk 1]
```diff
@@ -309,3 +312,25 @@ func namespacedViewName(v string, escaped bool) string {
...
+
+func newLabelDescriptors(keys []tag.Key) []*labelpb.LabelDescriptor {
+	labelDescriptors := make([]*labelpb.LabelDescriptor, len(keys))
+	for _, key := range keys {
+		labelDescriptors = append(
+			labelDescriptors,
+			&labelpb.LabelDescriptor{
+				Key:         key.Name(),
+				ValueType:   labelpb.LabelDescriptor_STRING, // We only use string tags
+				Description: "OpenCensus TagKey",
```
Comment (rakyll) [inline exporter/stats/stackdriver/stackdriver.go]: Do we need a description? If yes, I'd prefer we use "OpenCensus tag key". TagKey sounds programmatic.

File: exporter/stats/stackdriver/stackdriver_test.go [hunk 2]
```diff
@@ -35,7 +36,13 @@ func TestExporter_makeReq(t *testing.T) {
 	}
 	defer stats.DeleteMeasure(m)
 
-	cumView, err := stats.NewView("cumview", "desc", nil, m, stats.CountAggregation{}, stats.Cumulative{})
+	key, err := tag.NewKey("test-key")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tag.Delete(key)
```
Comment (rakyll) [inline exporter/stats/stackdriver/stackdriver_test.go]: This is not deleting the key but returning a mutator to delete a value keyed with key.

(FWIW, I think the naming for Insert, Upsert, Delete, etc is not great and creates confusion.)

File: exporter/stats/stackdriver/stackdriver.go (hunk 1 above)
Comment (songy23) [reply exporter/stats/stackdriver/stackdriver.go]: This is not a required filed, I think it's safe to omit it for now.

File: exporter/stats/stackdriver/stackdriver_test.go (hunk 2 above)
Comment (songy23) [reply exporter/stats/stackdriver/stackdriver_test.go]: Agree, this method is a little confusing to me at first. Removed.

File: exporter/stats/stackdriver/stackdriver.go [hunk 3]
```diff
@@ -309,3 +312,24 @@ func namespacedViewName(v string, escaped bool) string {
...
+func newLabels(tags []tag.Tag) map[string]string {
+	labels := make(map[string]string)
+	for _, tag := range tags {
+		labels[tag.Key.Name()] = tag.Value
+	}
+	return labels
+}
+
+func newLabelDescriptors(keys []tag.Key) []*labelpb.LabelDescriptor {
+	labelDescriptors := make([]*labelpb.LabelDescriptor, len(keys))
```
Comment (jcd2) [inline exporter/stats/stackdriver/stackdriver.go]: This is setting the slice size to len(keys), not its capacity.  So:
labelDescriptors := make([]*labelpb.LabelDescriptor, 0, len(keys))

File: exporter/stats/stackdriver/stackdriver.go [hunk 4]
```diff
@@ -309,3 +312,24 @@ func namespacedViewName(v string, escaped bool) string {
...
+	for _, key := range keys {
+		labelDescriptors = append(
+			labelDescriptors,
+			&labelpb.LabelDescriptor{
+				Key:       key.Name(),
```
Comment (jcd2) [inline exporter/stats/stackdriver/stackdriver.go]: The tag keys will need to be escaped for Stackdriver.  It limits length to 100 and only accepts some characters.  Letters, digits and underscores work.

Comment (songy23) [reply exporter/stats/stackdriver/stackdriver.go]: Sure, I added string sanitization in https://github.com/census-instrumentation/opencensus-go/pull/167

File: exporter/stats/stackdriver/stackdriver.go (hunk 3 above)
Comment (songy23) [reply exporter/stats/stackdriver/stackdriver.go]: Updated in https://github.com/census-instrumentation/opencensus-go/pull/167

**Pull request merged**